src/ui/WorkInProgressRoot.tsx

```diff
@@ -23,18 +23,21 @@ const CYCLES_PER_SEC = 1000 / CONSTANTS.MilliPerCycle;
 
 export function WorkInProgressRoot(): React.ReactElement {
   const setRerender = useState(false)[1];
+  const player = use.Player();
+  const router = use.Router();
   function rerender(): void {
     setRerender((old) => !old);
+    if (player.className != CONSTANTS.ClassStudyComputerScience && player.workType == CONSTANTS.WorkTypeStudyClass && player.money <= 0) {
```

Comment:
hmm after making more prs to the codebase i think doing this here may be a mistake....